drivers: tmc2209: use m0/m1-gpios instead of msx-gpios

jilaypandya opened the pull request:
1. Unify msx-gpios for step-dir drivers. tmc2209 is an exception for now and hence migrating to m0/m1-gpios instead of msx-gpios.
2. Generalize drv84xx tests for all step-dir drivers.
3. Added entry in migration-guide

jilaypandya committed:
drivers: tmc2209: use m0/m1-gpios instead of msx-gpios

unify msx-gpios for step-dir drivers. tmc2209 is an exception
for now and hence migrating to m0/m1-gpios instead of msx-gpios.

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

jilaypandya committed:
tests: generalize drv84xx tests for all step-dir drivers

drv84xx test-suite could be extended to other step-dir drivers
as well

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>

jbehrensnx commented on drivers/stepper/adi_tmc/tmc22xx/tmc22xx.c:
```diff
@@ -124,18 +139,16 @@ static int tmc22xx_stepper_init(const struct device *dev)
-			LOG_ERR("Failed to configure MSX pins: %d", ret);
-			return ret;
-		}
+	ret = tmc22xx_stepper_configure_msx_pins(dev);
```
If I understand this correctly, with these changes the init functions fails when both microstep pins are undefined, which is different from the current state.

jbehrensnx commented on dts/bindings/stepper/adi/adi,tmc2209.yaml:
```diff
@@ -25,13 +25,13 @@ properties:
-      The pins should be listed in the following order:
-      - MS1
-      - MS2
+    description: Microstep configuration pin MS1.
```
See `tmc22xx.c` concerning the pin names.

jbehrensnx replied:
Ah, the good old case of different vendors using different names for the same thing. With other words, there is no good solution. The pin names are M1 and M2, but the properties are m0 and m1 for consistency with the other drivers. Might possibly worth it to make a short note about this in the description of one of the pins to clear up the confusion, but this would be on the level of NIT for me.

jbehrensnx reviewed (changes requested):
First of all I approve standardization, through I am surprised that it goes in this direction, as I faintly remember a PR where drv84xx and the allegro driver were intended to be modified in the other direction.

The other thing is the test generalization which I feel should also be described in the PR description as an equal (if not possibly more important) part.

jilaypandya commented on tests/drivers/stepper/step_dir/src/main.c:
```diff
@@ -13,26 +13,67 @@
+	uint8_t m1: 2;
+};
+
+#if CONFIG_DRV84XX
```
Addressing another comment in the PR, i will move this to Kconfig. However, we would be then limited to testing only one particular micro-step resolution.

jilaypandya committed:
doc: add migration guide entry for replacing msx-gpios to m0/m1-gpios

msx-gpios is now replaced with m0/m1-gpios for adi,tmc2209 in order
to be consistent with other step/dir drivers

Signed-off-by: Jilay Pandya <jilay.pandya@outlook.com>
